tests/test_overviews.py

```diff
@@ -104,3 +106,23 @@ def test_build_overviews_new_file(tmpdir, path_rgb_byte_tif):
     with rasterio.open(dst_file, overview_level=1) as src:
         data = src.read()
         assert data.any()
+
+
+@requires_gdal33
+def test_ignore_overviews(data):
+    """Force ignore existing overviews when performing decimated read"""
+    inputfile = str(data.join('RGB.byte.tif'))
+    with rasterio.open(inputfile, 'r+') as src:
+        overview_shape = (3, math.ceil(src.height/2), math.ceil(src.width/2))
+        src.build_overviews([2], resampling=Resampling.nearest)
+        decimated_ovr = src.read(out_shape=overview_shape)
+
+    with rasterio.open(inputfile, OVERVIEW_LEVEL=-1) as src:
+        assert src.overviews(1) == []
+        assert src.overviews(2) == []
+        assert src.overviews(3) == []
+        decimated_avg = src.read(out_shape=overview_shape,
+                                 resampling=Resampling.average)
+
+    assert not np.array_equal(decimated_avg, decimated_ovr)
+
```

Comment:
@loicdtx I have an idea that could make testing more simple. We already have a https://github.com/mapbox/rasterio/blob/master/tests/test_boundless_read.py#L44 test case that uses difference in color between dataset and its overview to check for an .ovr hit. I think you could copy that test function into this module and then flip the sense of it around: checking the colors to ensure there is *no* .ovr hit when we add `OVERVIEW_LEVEL=-1` to https://github.com/mapbox/rasterio/blob/master/tests/test_boundless_read.py#L57. What do you think about that?

Reply:
Thanks for the review @sgillies. It took me some time to understand the logic and the pytest fixture, but after a bit of reading and digging I got it. See the improved test.